check50/__main__.py

```diff
@@ -159,6 +159,31 @@ def compile_checks(checks, prompt=False):
     return "__init__.py"
 
 
+def setup_logging(level):
+    """
+    Sets up logging for lib50.
+    level 'info' logs all git commands run to stderr
+    level 'debug' logs all git commands and their output to stderr
+    """
+    # No verbosity level set, don't log anything
+    if not level:
+        return
+
+    # Set verbosity level on the lib50 logger
+    lib50_logger = logging.getLogger("lib50")
+    if level == "debug":
+        lib50_logger.setLevel(logging.DEBUG)
+    elif level == "info":
+        lib50_logger.setLevel(logging.INFO)
+    else:
+        raise ValueError(f'Unknown logging level: "{level}"')
```

Comment:
Just for flexibility I think, especially that it doesn't cost much to add? I don't think it's unreasonable for someone to pass `error`, `warning`, etc and expect it to work since we're using the standard logging module. Besides, if we ever decide to use any of these methods in the future, we wouldn't have to change this code.

Reply:
Oof, that example in the Python docs is awfully hacky! Turns out there are multiple booleans in `logging` such as `logMultiprocessing` that are also instances of `int`. The only reason you can't set `--log-level=logMultiprocessing` is because they happen to set `.upper()` for different reasons:

```
# Convert to upper case to allow the user to
# specify --log=DEBUG or --log=debug
```

Reply:
Couple curiosities here:

We would need to rely on `logging` to not add any more `int` constants in ALL CAPS. But if their docs rely on that, so can we I guess.

Previously setting any `--verbose` level would imply `--log`. This made sense because everything you could set was strictly more verbose than the default. However now, for some reason, you could set `--verbose ERROR`. In which case your intent is probably to show less. Given that this option now exists, I removed the implication of `--log` altogether.

Not sure this is worth it tbh. @crossroads1112 thoughts?

Reply:
``` 
not instance(level, bool) and isinstance(level, int)
```
?

Reply:
Alternatively, we could use `getLevelName`? https://docs.python.org/3/library/logging.html#logging.getLevelName

Reply:
`not isinstance(level, bool) and isinstance(level, int)`

Not sure what this fixes 😅

`getLevelName`, I could run through 0 to 10 I guess to get all level names, but... that doesn't feel any less hacky?

Reply:
> Not sure what this fixes 

I thought the problem was that there were `bool` attributes in `logging` that returned `True` when you did `isinstance(level, int)` since a `isinstance(False, int)` is `True` in Python in which case that would exclude `bool`s at least?

```
>>> level = False
>>> not isinstance(level, bool) and isinstance(level, int)
False
>>> level = 3
>>> not isinstance(level, bool) and isinstance(level, int)
True
```

Reply:
@Jelleas why would you need to run through levels 0 to 10? In this case you'd just pass the uppercase of the name passed and check if the return value is `int` and use it?

Reply:
It's nice to be able to enumerate all possible options in argparse. Not only do we then get the help message, but also validation for free.

Reply:
I get it, but there's no need for `not isinstance(level, bool) and isinstance(level, int)`. I just used what the docs used: `isinstance(level, int) and level.isupper()`

Reply:
Oh gotcha. The catch is, it'd be better to not assume anything about the underlying integers (e.g., that they're in 0 to 10). We shouldn't really care about what they are. Would pasting https://docs.python.org/3/library/logging.html#logging-levels in the help make sense perhaps?

Reply:
Well not entirely, we wouldn't get validation that way, unless we build it ourselves. I'm also not a fan of linking externally to explain something we can easily explain ourselves.

Reply:
Zoom out a little bit. We're trying to solve a problem for a feature that adds zero value currently and unlikely will in the future. Every solution, from `getLevelName`, to `isinstance` all the things, to accessing `_levelNames` is hacky in one way or another. 

I'd say this trade off, for easier potential future extensibility vs complexity and hacky code, is not worth it. Especially since we can add support whenever we do actually need it.

If you insist though, lets go with whatever the module prescribes, which in this case is `isinstance and .isupper()`.

Reply:
Ugh, that's annoying. Alright, it's probably not worth it like you said.